src/deepsparse/utils/state.py

```diff
@@ -43,9 +46,23 @@ def create_state(self, new_state: dict):
         if self._current_state:
             raise ValueError("State creation is only allowed during initialization.")
         self._current_state = new_state
+        
+class TimerState:
+    def __init__(self, timer: Optional[Timer] = None):
+        self.timer = timer
+        
+    @contextmanager
+    def record(self, id: str):
```

Comment:
prefer to rename this to `time()` especially since it's being called from inference state directly